The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11088951227/job/30809475639) [(plain)](https://github.com/rust-lang/rust/commit/6683c0416cee4e0724393749881ddf0c15bc50ef/checks/30809475639/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    assertion `left == right` failed: NOP unsize vtable changed principal trait ref: for<'a> Fn(&'a i32) -> &i32 -> Fn(&i32) -> &i32
      left: Some(Binder { value: std::ops::Fn<(&i32,)>, bound_vars: [Region(BrAnon)] })
     right: Some(Binder { value: std::ops::Fn<(&i32,)>, bound_vars: [] })
    stack backtrace:
       0:     0x7f7922798ba1 - <<std[f5b58013ef4c310a]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[9b60264776027f33]::fmt::Display>::fmt
       1:     0x7f79227f5494 - core[9b60264776027f33]::fmt::write
       2:     0x7f792278bd49 - <std[f5b58013ef4c310a]::sys::pal::unix::stdio::Stderr as std[f5b58013ef4c310a]::io::Write>::write_fmt
       3:     0x7f7922798a42 - <std[f5b58013ef4c310a]::sys::backtrace::BacktraceLock>::print
       4:     0x7f792279b4ca - std[f5b58013ef4c310a]::panicking::default_hook::{closure#1}
       5:     0x7f792279b25d - std[f5b58013ef4c310a]::panicking::default_hook
       6:     0x7f791ded7162 - <alloc[b702644163b63f59]::boxed::Box<rustc_driver_impl[5bfc79a9a274d93]::install_ice_hook::{closure#0}> as core[9b60264776027f33]::ops::function::Fn<(&dyn for<'a, 'b> core[9b60264776027f33]::ops::function::Fn<(&'a std[f5b58013ef4c310a]::panic::PanicHookInfo<'b>,), Output = ()> + core[9b60264776027f33]::marker::Send + core[9b60264776027f33]::marker::Sync, &std[f5b58013ef4c310a]::panic::PanicHookInfo)>>::call
       7:     0x7f792279bda8 - std[f5b58013ef4c310a]::panicking::rust_panic_with_hook
       8:     0x7f792279bace - std[f5b58013ef4c310a]::panicking::begin_panic_handler::{closure#0}
       9:     0x7f79227991f9 - std[f5b58013ef4c310a]::sys::backtrace::__rust_end_short_backtrace::<std[f5b58013ef4c310a]::panicking::begin_panic_handler::{closure#0}, !>
      11:     0x7f79227f1310 - core[9b60264776027f33]::panicking::panic_fmt
      12:     0x7f79227f1866 - core[9b60264776027f33]::panicking::assert_failed_inner
      12:     0x7f79227f1866 - core[9b60264776027f33]::panicking::assert_failed_inner
      13:     0x7f791e56750f - core[9b60264776027f33]::panicking::assert_failed::<core[9b60264776027f33]::option::Option<rustc_type_ir[ee8249714c75c8c0]::binder::Binder<rustc_middle[a3ae3505e09a5e91]::ty::context::TyCtxt, rustc_type_ir[ee8249714c75c8c0]::predicate::ExistentialTraitRef<rustc_middle[a3ae3505e09a5e91]::ty::context::TyCtxt>>>, core[9b60264776027f33]::option::Option<rustc_type_ir[ee8249714c75c8c0]::binder::Binder<rustc_middle[a3ae3505e09a5e91]::ty::context::TyCtxt, rustc_type_ir[ee8249714c75c8c0]::predicate::ExistentialTraitRef<rustc_middle[a3ae3505e09a5e91]::ty::context::TyCtxt>>>>
      14:     0x7f791e4d9341 - rustc_codegen_ssa[4c0572b4d8f5b39a]::base::unsized_info::<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>
      15:     0x7f791e4d8ac7 - rustc_codegen_ssa[4c0572b4d8f5b39a]::base::unsize_ptr::<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>
      16:     0x7f791e475cb9 - <rustc_codegen_ssa[4c0572b4d8f5b39a]::mir::FunctionCx<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>>::codegen_rvalue_operand
      17:     0x7f791e46a5d0 - rustc_codegen_ssa[4c0572b4d8f5b39a]::mir::codegen_mir::<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>
      18:     0x7f791e4dad94 - rustc_codegen_ssa[4c0572b4d8f5b39a]::base::codegen_instance::<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>
      19:     0x7f791e3f73a2 - <rustc_middle[a3ae3505e09a5e91]::mir::mono::MonoItem as rustc_codegen_ssa[4c0572b4d8f5b39a]::mono_item::MonoItemExt>::define::<rustc_codegen_llvm[fc8d7ca60168e96]::builder::Builder>
      20:     0x7f791e57e396 - rustc_codegen_llvm[fc8d7ca60168e96]::base::compile_codegen_unit::module_codegen
      21:     0x7f791e58130f - <rustc_codegen_llvm[fc8d7ca60168e96]::LlvmCodegenBackend as rustc_codegen_ssa[4c0572b4d8f5b39a]::traits::backend::ExtraBackendMethods>::compile_codegen_unit
      22:     0x7f791e4da1fd - rustc_codegen_ssa[4c0572b4d8f5b39a]::base::codegen_crate::<rustc_codegen_llvm[fc8d7ca60168e96]::LlvmCodegenBackend>
      23:     0x7f791e584bb9 - <rustc_codegen_llvm[fc8d7ca60168e96]::LlvmCodegenBackend as rustc_codegen_ssa[4c0572b4d8f5b39a]::traits::backend::CodegenBackend>::codegen_crate
      24:     0x7f791e1515b7 - <rustc_session[f3c16d4c488ca246]::session::Session>::time::<alloc[b702644163b63f59]::boxed::Box<dyn core[9b60264776027f33]::any::Any>, rustc_interface[355e18af254c6911]::passes::start_codegen::{closure#0}>
      25:     0x7f791e2ef717 - rustc_interface[355e18af254c6911]::passes::start_codegen
      26:     0x7f791e1852a2 - <rustc_interface[355e18af254c6911]::queries::Linker>::codegen_and_build_linker
      27:     0x7f791defa0ec - <rustc_interface[355e18af254c6911]::queries::QueryResult<&rustc_middle[a3ae3505e09a5e91]::ty::context::GlobalCtxt>>::enter::<core[9b60264776027f33]::result::Result<core[9b60264776027f33]::option::Option<rustc_interface[355e18af254c6911]::queries::Linker>, rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}::{closure#1}::{closure#6}>
      28:     0x7f791de70162 - <rustc_interface[355e18af254c6911]::interface::Compiler>::enter::<rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}::{closure#1}, core[9b60264776027f33]::result::Result<core[9b60264776027f33]::option::Option<rustc_interface[355e18af254c6911]::queries::Linker>, rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>
      29:     0x7f791df44225 - <scoped_tls[f263722640ea45a5]::ScopedKey<rustc_span[77f251a5b4c0f966]::SessionGlobals>>::set::<rustc_interface[355e18af254c6911]::util::run_in_thread_with_globals<rustc_interface[355e18af254c6911]::util::run_in_thread_pool_with_globals<rustc_interface[355e18af254c6911]::interface::run_compiler<core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}>::{closure#1}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}::{closure#0}::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>
      30:     0x7f791dee59cc - rustc_span[77f251a5b4c0f966]::create_session_globals_then::<core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_interface[355e18af254c6911]::util::run_in_thread_with_globals<rustc_interface[355e18af254c6911]::util::run_in_thread_pool_with_globals<rustc_interface[355e18af254c6911]::interface::run_compiler<core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}>::{closure#1}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}::{closure#0}::{closure#0}>
      31:     0x7f791df3a2a2 - std[f5b58013ef4c310a]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[355e18af254c6911]::util::run_in_thread_with_globals<rustc_interface[355e18af254c6911]::util::run_in_thread_pool_with_globals<rustc_interface[355e18af254c6911]::interface::run_compiler<core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}>::{closure#1}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>
      32:     0x7f791df50c3f - <<std[f5b58013ef4c310a]::thread::Builder>::spawn_unchecked_<rustc_interface[355e18af254c6911]::util::run_in_thread_with_globals<rustc_interface[355e18af254c6911]::util::run_in_thread_pool_with_globals<rustc_interface[355e18af254c6911]::interface::run_compiler<core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>, rustc_driver_impl[5bfc79a9a274d93]::run_compiler::{closure#0}>::{closure#1}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[9b60264776027f33]::result::Result<(), rustc_span[77f251a5b4c0f966]::ErrorGuaranteed>>::{closure#1} as core[9b60264776027f33]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
      33:     0x7f79227a7b55 - <std[f5b58013ef4c310a]::sys::pal::unix::thread::Thread>::new::thread_start
      35:     0x7f791bf49850 - <unknown>
      36:                0x0 - <unknown>
    
    error: the compiler unexpectedly panicked. this is a bug.
---
......FFFF (130/131)
F          (131/131)


/checkout/tests/rustdoc-gui/scrape-examples-layout.goml scrape-examples-layout... FAILED
[ERROR] `tests/rustdoc-gui/scrape-examples-layout.goml` line 93: The following errors happened: [different Y values: 287 (or 287) != 281]: for command `assert-position: (".scraped-example", {"y": 281})`
[ERROR] `tests/rustdoc-gui/scrape-examples-layout.goml` line 94: The following errors happened: [different Y values: 315 (or 315) != 309]: for command `assert-position: (".scraped-example .prev", {"y": 281 + |offset_y| + |title_height|})`
/checkout/tests/rustdoc-gui/search-result-display.goml search-result-display... FAILED
/checkout/tests/rustdoc-gui/search-result-display.goml search-result-display... FAILED
[WARNING] `tests/rustdoc-gui/search-result-display.goml` line 37: Delta is 0 for "x", maybe try to use `compare-elements-position` instead?

[ERROR] `tests/rustdoc-gui/search-result-display.goml` line 67: The following errors happened: [expected a width of `509`, found `370`]: for command `assert-size: ("#crate-search", {"width": 509})`

/checkout/tests/rustdoc-gui/sidebar-source-code-display.goml sidebar-source-code-display... FAILED
[ERROR] `tests/rustdoc-gui/sidebar-source-code-display.goml` line 145: The following errors happened: [expected `2564` for window property `pageYOffset`, found `2570`]: for command `assert-window-property: {"pageYOffset": |y_offset|}`
[ERROR] `tests/rustdoc-gui/sidebar-source-code-display.goml` line 152: The following errors happened: [expected `2564` for window property `pageYOffset`, found `2570`]: for command `assert-window-property: {"pageYOffset": |y_offset|}`

/checkout/tests/rustdoc-gui/source-anchor-scroll.goml source-anchor-scroll... FAILED
[ERROR] `tests/rustdoc-gui/source-anchor-scroll.goml` line 11: The following errors happened (for selector `html`): [expected `194` for property `"scrollTop"`, found `200`]: for command `assert-property: ("html", {"scrollTop": "194"})`
[ERROR] `tests/rustdoc-gui/source-anchor-scroll.goml` line 13: The following errors happened (for selector `html`): [expected `225` for property `"scrollTop"`, found `231`]: for command `assert-property: ("html", {"scrollTop": "225"})`
[ERROR] `tests/rustdoc-gui/source-anchor-scroll.goml` line 15: The following errors happened (for selector `html`): [expected `122` for property `"scrollTop"`, found `128`]: for command `assert-property: ("html", {"scrollTop": "122"})`
[ERROR] `tests/rustdoc-gui/source-anchor-scroll.goml` line 20: The following errors happened (for selector `html`): [expected `122` for property `"scrollTop"`, found `128`]: for command `assert-property: ("html", {"scrollTop": "122"})`
/checkout/tests/rustdoc-gui/toggle-docs-mobile.goml toggle-docs-mobile... FAILED
/checkout/tests/rustdoc-gui/toggle-docs-mobile.goml toggle-docs-mobile... FAILED
[ERROR] `tests/rustdoc-gui/toggle-docs-mobile.goml` line 7: The following errors happened (for selector `.top-doc`): [assert didn't fail for attribute `open` (``)]: for command `assert-attribute-false: (".top-doc", {"open": ""})`
[ERROR] `tests/rustdoc-gui/toggle-docs-mobile.goml` line 28: The following errors happened (for selector `.top-doc`): [assert didn't fail for attribute `open` (``)]: for command `assert-attribute-false: (".top-doc", {"open": ""})`
Error: ()
Build completed unsuccessfully in 0:03:30
  local time: Sun Sep 29 03:22:22 UTC 2024
  network time: Sun, 29 Sep 2024 03:22:22 GMT
```

</details>

--- a/tests/rustdoc-gui/scrape-examples-layout.goml
+++ b/tests/rustdoc-gui/scrape-examples-layout.goml
@@ -90,8 +90,8 @@ assert-css: (".scraped-example .example-wrap::after", {"bottom": "0px"})
 // Then with mobile
 set-window-size: (600, 600)
 store-size: (".scraped-example .scraped-example-title", {"height": title_height})
-assert-position: (".scraped-example", {"y": 281})
-assert-position: (".scraped-example .prev", {"y": 281 + |offset_y| + |title_height|})
+assert-position: (".scraped-example", {"y": 287})
+assert-position: (".scraped-example .prev", {"y": 287 + |offset_y| + |title_height|})
 
 define-function: (
     "check_title_and_code_position",

--- a/tests/rustdoc-gui/search-result-display.goml
+++ b/tests/rustdoc-gui/search-result-display.goml
@@ -64,7 +64,7 @@ set-text: (
 )
 
 // Then we compare again to confirm the height didn't change.
-assert-size: ("#crate-search", {"width": 509})
+assert-size: ("#crate-search", {"width": 370})
 assert-size: (".search-results-title", {
     "height": |search_results_title_height|,
 })

--- a/tests/rustdoc-gui/sidebar-source-code-display.goml
+++ b/tests/rustdoc-gui/sidebar-source-code-display.goml
@@ -141,7 +141,7 @@ click: "#sidebar-button"
 wait-for-css: (".src .sidebar > *", {"visibility": "hidden"})
 // We scroll to line 117 to change the scroll position.
 scroll-to: '//*[@id="117"]'
-store-value: (y_offset, "2564")
+store-value: (y_offset, "2570")
 assert-window-property: {"pageYOffset": |y_offset|}
 // Expanding the sidebar...
 click: "#sidebar-button"

--- a/tests/rustdoc-gui/source-anchor-scroll.goml
+++ b/tests/rustdoc-gui/source-anchor-scroll.goml
@@ -8,13 +8,13 @@ set-window-size: (600, 800)
 assert-property: ("html", {"scrollTop": "0"})
 
 click: '//a[text() = "barbar" and @href="#5-7"]'
-assert-property: ("html", {"scrollTop": "194"})
+assert-property: ("html", {"scrollTop": "200"})
 click: '//a[text() = "bar" and @href="#28-36"]'
-assert-property: ("html", {"scrollTop": "225"})
+assert-property: ("html", {"scrollTop": "231"})
 click: '//a[normalize-space() = "sub_fn" and @href="#2-4"]'
-assert-property: ("html", {"scrollTop": "122"})
+assert-property: ("html", {"scrollTop": "128"})
 
 // We now check that clicking on lines doesn't change the scroll
 // Extra information: the "sub_fn" function header is on line 1.
 click: '//*[@id="6"]'
-assert-property: ("html", {"scrollTop": "122"})
+assert-property: ("html", {"scrollTop": "128"})

--- a/tests/rustdoc-gui/toggle-docs-mobile.goml
+++ b/tests/rustdoc-gui/toggle-docs-mobile.goml
@@ -3,12 +3,12 @@
 go-to: "file://" + |DOC_PATH| + "/test_docs/struct.Foo.html"
 set-window-size: (433, 600)
 assert-attribute: (".top-doc", {"open": ""})
-click: (4, 260) // This is the position of the top doc comment toggle
+click: (4, 270) // This is the position of the top doc comment toggle
 assert-attribute-false: (".top-doc", {"open": ""})
-click: (4, 260)
+click: (4, 270)
 assert-attribute: (".top-doc", {"open": ""})
 // To ensure that the toggle isn't over the text, we check that the toggle isn't clicked.
-click: (3, 260)
+click: (3, 270)
 assert-attribute: (".top-doc", {"open": ""})
 
 // Assert the position of the toggle on the top doc block.
@@ -24,10 +24,10 @@ assert-position: (
 // Now we do the same but with a little bigger width
 set-window-size: (600, 600)
 assert-attribute: (".top-doc", {"open": ""})
-click: (4, 260) // New Y position since all search elements are back on one line.
+click: (4, 270) // New Y position since all search elements are back on one line.
 assert-attribute-false: (".top-doc", {"open": ""})
-click: (4, 260)
+click: (4, 270)
 assert-attribute: (".top-doc", {"open": ""})
 // To ensure that the toggle isn't over the text, we check that the toggle isn't clicked.
-click: (3, 260)
+click: (3, 270)
 assert-attribute: (".top-doc", {"open": ""})

